### ISSUE TITLE
Switching between audio/video devices

### ISSUE THREAD

**Issue by ostasevych** [bigbluebutton/bigbluebutton#9723]
Quite often a user may have several audio devices on one's device, eg  on the smartphone: earphones and dynamic, and laptop: earphones, BT dynamic. 

The same may be with web cameras, especially on smartphones

Please, add the switcher to switch between audio and video devices on the fly, eg, lik Jitsi Meed has done:
![image](https://user-images.githubusercontent.com/5199030/83329574-f48c9400-a292-11ea-844e-2154eb25e1ca.png)

**Comment by narayanshivansh49** [comment]
Yup this will be a great feature. I think we are getting this in 2.3 not sure.

**Comment by ostasevych** [comment]
> Yup this will be a great feature. I think we are getting this in 2.3 not sure.

Great, you know, I am not so happy with the way how Jitsi did that. I even suggest, to add a new button to switch between devices depending on available audio devices: eg headphones plugged in the jack, dynamic of the device and Bluetooth device, if connected. So, you may click on the button and select the device circularly, and the icon of the button will be changing. 

The same may be applied to the microphone (internal and external) and the camera (front or back). So, to have 3 dynamic buttons for 2 audio devices, and 1 video device, which show their state.

**Comment by jsduffy** [comment]
This would be awesome!

**Comment by mired42** [comment]
Having worked with both BBB and Jitsi for month now and having seen the recent addition of the "quick device selection" for both audio and video (not shown above), I have to say that I really appreciate Jitsi's solution. Also, the feedback from our users has been very good. Users actually find it easier now to configure their AV devices. I can highly recommend this and I really hope to see something like this in BBB as well .

**Comment by antobinary** [comment]
This is being worked on in https://github.com/bigbluebutton/bigbluebutton/pull/10129 (just linking the PR to the issue)

**Comment by mariogasparoni** [comment]
This one can be closed by #11813

**Comment by antobinary** [comment]
I changed the title to be only audio related. Both the screenshot and the PRs relating to it are audio-only.

It would make sense to have similar work done for webcams, but it is different since we technically support multiple webcams per person. We should discuss a bit whether it makes sense to try to do something like this for webcams, or stick to the existing behaviour
